doc: Improve CI docs on env and qemu-user-static

hodlinator opened the pull request:
Should at least partially fix #31199

hodlinator committed:
doc: CI - Describe qemu-user-static usage

Should help in cases such as: https://github.com/bitcoin/bitcoin/pull/31144#issuecomment-2450578651

maflcko reviewed (approved):
lgtm ACK a8afb65ce9b8cd5fe90fcc570087f175c3662cad

nit: Not sure if the commit message is fully correct

waketraindev commented:
Concept ACK

nit: Please squash the commits per the contributing guidelines: https://github.com/bitcoin/bitcoin/blob/master/CONTRIBUTING.md#squashing-commits

Both commits touch `ci/README.md` and cover the same scope.

janb84 commented on ci/README.md:
```diff
@@ -28,8 +28,18 @@ the CI locally you may need to reduce that entropy by running:
+
+It is recommended to run the CI system in a clean environment. `env -i` below
+ensures *only* specified environment variables are propagated into the local CI.
+Run the test stage with a specific configuration:
```
```suggestion
It is recommended to run the CI system in a clean environment. The `env -i` command
below ensures that *only* specified environment variables are propagated into the local CI.
To run the test stage with a specific configuration:
```

janb84 reviewed (commented):
Concept ACK f8d83ee4d631c95ef6e222d7868bfa9e28b66e05

This PR improves the CI documentation, to clarify certain aspects of the CI which raise questions time to time. (see #31199). I would say that the additional information in this PR would stop some questions from being asked. The information provided is (the docker part) is non-trivial to me, and the importance of `env-i` can be overlooked so the additional information is welcome.  

NIT: Small grammar suggestion.

hodlinator committed:
doc: CI - Clarify how important `env -i` is and why

Would hopefully have helped me in this case: https://github.com/bitcoin/bitcoin/pull/31176#issuecomment-2526410039
Since then however, fd813bf863b1ffa91429de6342285b35bab2bfa4 also made sure a minimal environment is used.

janb84 reviewed (commented):
ACK 552eb90071fd246ba40037f74329403b72453047

changes since last ACK: 
- grammar changes.

Thanks for incorporating my NIT.
